src/charm.py

```diff
@@ -16,13 +21,213 @@ class GraylogCharm(CharmBase):
 
     def __init__(self, *args):
         super().__init__(*args)
+
+        # initialize image resource
+        self.image = OCIImageResource(self, 'graylog-image')
+
+        # event observations
         self.framework.observe(self.on.config_changed, self._on_config_changed)
+        self.framework.observe(self.on.update_status, self._on_update_status)
+        self.framework.observe(self.on.stop, self._on_stop)
+        self.framework.observe(
+            self.on['elasticsearch'].relation_changed,
+            self._on_elasticsearch_relation_changed
+        )
+        self.framework.observe(
+            self.on['elasticsearch'].relation_broken,
+            self._on_elasticsearch_relation_broken
+        )
+        self.framework.observe(
+            self.on["mongodb"].relation_changed, self._on_mongodb_relation_changed
+        )
+        self.framework.observe(
+            self.on["mongodb"].relation_broken, self._on_mongodb_relation_broken
+        )
+        # initialized stored variables
+        self._stored.set_default(elasticsearch_uri=str())  # connection str for elasticsearch
+        self._stored.set_default(mongodb_uri=str())  # connection info for mongodb
+        self._stored.set_default(password_secret=str())
+
+    @property
+    def bind_address(self):
+        """The HTTP bind address used for the web interface"""
+        # TODO: come up with a more consistent way of doing this
+        addr = 'graylog-0.graylog-endpoints'
```

Comment:
Is this an API endpoint for incoming requests ? If so why is this a hostname ? Will `0.0.0.0` suffice instead ?